src/test/integration/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandlerProviderImplIT.java

```diff
@@ -201,12 +210,56 @@ public void shouldCreateChargeTransaction() {
         assertThat(chargeTxn.getState()).isEqualTo(SUCCESS);
     }
 
+    @Test
+    public void whenCtpPaymentIsAlreadyMarkedAsSuccess_shoudNotCallCreateTransactionInPaypalPlus() {
+
+        final CtpPaymentWithCart ctpPaymentWithCart = createCartWithPayment();
+        final PaymentHandler paymentHandler = paymentHandlerProvider
+                .getPaymentHandler(MAIN_TEST_TENANT_NAME)
+                .get();
+
+        io.sphere.sdk.payments.Payment payment = ctpPaymentWithCart.getPayment();
+
+        final TransactionDraft transactionDraft = TransactionDraftBuilder
+                .of(CHARGE, Money.of(200, EUR)).state(TransactionState.SUCCESS)
+                .build();
+        final AddTransaction addTransaction = AddTransaction
+                .of(transactionDraft);
+        final SetInterfaceId interfaceIdUpdateAction = SetInterfaceId.of("testInterfaceId");
+        List<UpdateAction<io.sphere.sdk.payments.Payment>> actions = new ArrayList<>();
+        actions.add(addTransaction);
+        actions.add(interfaceIdUpdateAction);
+
+        final io.sphere.sdk.payments.Payment updatedPayment = this
+                .sphereClient
+                .execute(PaymentUpdateCommand.of(payment, actions))
+                .toCompletableFuture()
+                .join();
+
+        final String ctpPaymentId = updatedPayment.getId();
+
+        executeBlocking(paymentHandler.createPayment(ctpPaymentId));
+
+        PaymentHandleResponse response = executeBlocking(paymentHandler
+                .executePayment(updatedPayment.getInterfaceId(), "testPayerId"));
+
+        io.sphere.sdk.payments.Payment ctpPayment = executeBlocking(sphereClient
+                .execute(PaymentByIdGet.of(ctpPaymentId)));
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED.value());
```

Comment:
How do you know from this test that it was not created when running `executePayment` method?

Reply:
It's from the number of interactions count

Reply:
Yes, and it has 1 😄 so it was created during `executePayment`?

Reply:
Add a verification that the number of interface interactions count didn't change after you run the execute payment.

Reply:
Why do you rely on count of interactions and not count of transactions? Change prevents obsolete transaction creation and not interaction.

Reply:
0f6df796760637e1ed5d99a1b49dc13e078a07c8